### PR TITLE
Implementing new design system heading typography in the code base

### LINKED ISSUE #1743: Implement Design system heading typography in the code base
### Overview
We need to implement the headings part of the design system as classes in the code base so that our heading sizes are no longer attached to h1/h2/h3 tags etc and can no longer negatively affect the accessibility/hierarchy of the web pages for people using screen readers

### Action Items

- [x] Go to the [Figma Design System page](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=3464%3A3) Typography section on the top row
- [x] Read the Start Here if you do not know about design systems (_*Danielle note: sorry, the developer summary isn't there yet! But the designer one still explains the basics_)
- [x] Find the headings (from Title1 to Title6 ) in the figma link (See 1.Style & Rules => Typography)
- [x] Add new classes Title1 through Title6 with only the size properties (not the other properties like typeface, line height etc. at this time because that may be automatically applied to all typography on the site elsewhere) into the code base in the appropriate scss file (the classes will be implemented in headings throughout the site in a future issue in collaboration with design, after this step is finished)

Screen Shot 2021-06-17 at 3 43 02 PM

### Resources/Instructions
[Figma](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=3464%3A3)

### PR DESCRIPTION
**Pull request by Zak234**
Fixes #1743 

Added six new classes of heading: Title1, Title2, Title3, Title4, Title5 and Title6. Each one was only given size attributes as specified by the issue. These have not been applied yet to any text as this will be done in a different issue. The goal of these changes was so that our heading sizes are no longer attached to h1/h2/h3 tags etc and can no longer negatively affect the accessibility/hierarchy of the web pages for people using screen readers

### CODE REVIEW

**Comment by akibrhast** [comment]
Update: This will be reviewed and potentially discussed in tonight's backend meeting

File: _sass/elements/_typography.scss [hunk 1]
```diff
@@ -63,6 +71,42 @@ h4,
...
+  font: map-get($font-weight, bold) #{map-get($font-size-Title4, mobile)}
+  map-get($font-family, decorative);
+}
+
+Title5
+.type-Title5 {
+  font: map-get($font-weight, bold) #{map-get($font-size-Title5, mobile)}
+  map-get($font-family, decorative);
+}
+
```
Comment (Aveline-art) [inline _sass/elements/_typography.scss]: From the pattern of the code above and below, it looks like a comma was missed between the tag and the class.

File: _sass/variables/_typography.scss [hunk 2]
```diff
@@ -69,6 +69,48 @@ $font-size-h1: (
   /* 40px*/ desktop: 3rem /* 48px*/
 );
 
+$font-size-Title1: (
+  print: 36pt,
+  /* 32px*/ mobile: 2rem,
+  /* 40px*/ tablet: 2.5rem,
+  /* 48px*/ desktop: 3rem
+);
```
Comment (Aveline-art) [inline _sass/variables/_typography.scss]: Out of curiosity, where were the print values pulled from? I could not find the 36pt et al. from the Figma design.

**Comment by akibrhast** [comment]
@Zak234  Great work on this! it might take a while to get reviewed and merged. In the mean time, maybe pick up another large issue or something from the engre CoP . Or, if you want something interesting and juicy hit me up and we will try to find something for you to sink your teeth into!

### Update
This was discussed and and reviewed in group during the thursday meeting. Majority approval in that everything looks good and this should not break anything. However individual developers assigned to review this issue has yet to test this out locally on their systems. 

- The people tagged on this issue could you all please review the changes with respect to the design system as linked in the issue, and ensure that none of the existing pages has any weird behaviour(It shouldn't).

- Also please double check the compiled css of these title classes that is generated within the `_site` directory and ensure that everything is compiling down to as expected.

**Review by Aveline-art — COMMENTED** [review]
Hey @Zak234! Overall, great work on adding the new typography and maintaining the structure of the website. The team had reviewed your work in the backend meeting and we are very impressed with your speed and accuracy. Please check my comments (which are my own, not the team's). Once they have been addressed, I can begin to test out the new typography!

Note to reviewers: I have spoken to designers about the print sizes and there are currently no plans to do anything with them, in terms of standardizing. You might say, the way that they are here are mostly fine/inconsequential. I am not sure if we would want print sizes to be a part of the changes, but I am also not sure a discussion is useful. Perhaps we can quickly decide whether we want them at all?

File: _sass/elements/_typography.scss (hunk 1 above)
Comment (Zak234) [reply _sass/elements/_typography.scss]: Done

File: _sass/variables/_typography.scss (hunk 2 above)
Comment (Zak234) [reply _sass/variables/_typography.scss]: I calculated it from the desktop px size.

**Review by akibrhast — CHANGES_REQUESTED** [review]
- Add . infront of the the title class
- make the title class names lower cases , example: `.title1`

File: _sass/elements/_typography.scss [hunk 3]
```diff
@@ -180,6 +255,37 @@ map-get($font-family, body);
     line-height: #{map-get($line-height-h4, desktop)};
   }
 
+  title1,
+  .type-Title1 {
+    font-size: #{map-get($font-size-title1, desktop)};
+  }
+
```
Comment (Aveline-art) [inline _sass/elements/_typography.scss]: ".type-Title" is still uppercase (JS-case? camelcase?), here and elsewhere!

**Review by Aveline-art — CHANGES_REQUESTED** [review]
I have checked out the new design on some of the pages and it looks awesome! Thank you for doing this @Zak234 . There is only one fix left before I approve of this. Cannot wait to start implementing this everywhere.

File: _sass/elements/_typography.scss (hunk 3 above)
Comment (Zak234) [reply _sass/elements/_typography.scss]: Done

**Review by Aveline-art — APPROVED** [review]
Everything looks good! Great work!

**Review by jbubar — APPROVED** [review]
I pulled this to my machine, everything looks great. Sakari, Nice job. Stelllar.

**Pull request merged**